### PR TITLE
CONSOLE-3077: Promote ConsolePlugins API version to v1 in console repository

### PR DESCRIPTION
**Pull request by jhadvig**
Story: https://issues.redhat.com/browse/CONSOLE-3077

Updating ConsolePlugin manifest, model and type in console and also Readme docs.
Link to the v1 api for ConsolePlugin CRD - https://github.com/openshift/api/blob/master/console/v1/types_console_plugin.go

/assign @TheRealJon

### CODE REVIEW

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
QE Approver: /assign @zhaozhanqi 
Docs Approver: /assign @opayne1
PX Approver:	 /assign @RickJWagner

**Comment by jhadvig** [comment]
QE Approver:
 /assign @zhaozhanqi
Docs Approver:
 /assign @opayne1
PX Approver:
 /assign @RickJWagner

File: dynamic-demo-plugin/README.md [hunk 1]
```diff
@@ -183,12 +186,35 @@ conster Header: React.FC = () => {
...
-the i18n namespace named after the dynamic plugin, in this case `plugin__console-demo-plugin`,
-are loaded. If the annotation is set to any other value or is missing on the `ConsolePlugin`
-resource, localization resources are not loaded.
+To indicate wheather the `ConsolePlugin` contains localization resources
+set `spec.i18n.loadType` field based on needed behaviour. `Preload` will
+load all plugin's localization resources from the i18n namespace named
+after the dynamic plugin, in this case `plugin__console-demo-plugin`,
+during loading of the plugin. If set to `Lazy` plugin's localization
+resources wont be preload, instead will leave thier loading to runtime's
+lazy-loading.
```
Comment (opayne1) [inline dynamic-demo-plugin/README.md:195]: Just a few spelling and grammar nits on this one.

```suggestion

To indicate whether the `ConsolePlugin` contains localization resources,
set `spec.i18n.loadType` field based on needed behavior. `Preload` will
load all plugin's localization resources from the i18n namespace named
after the dynamic plugin during loading. In this case, `plugin__console-demo-plugin`.
If set to `Lazy`, the plugin's localization resources wont be preload, instead will leave their loading to runtime's
lazy-loading.
```

**Review by opayne1 — COMMENTED** [review]
Just a few minor spelling/grammar suggestions. :)

**Comment by jhadvig** [comment]
@opayne1 thanks for review. Comment addressed :)

File: frontend/public/module/k8s/types.ts [hunk 2]
```diff
@@ -1113,12 +1113,33 @@ export type NetworkPolicyPort = {
...
+      service: {
+        basePath: string;
+        name: string;
+        namespace: string;
+        port: number;
+      };
+      type: 'Service';
+    };
+    i18n: {
+      loadType: 'Preload' | 'Lazy' | '';
```
Comment (yapei) [inline frontend/public/module/k8s/types.ts:1126]: `''` empty string is not valid `loadType` value

Comment (yapei) [reply frontend/public/module/k8s/types.ts:1126]: // LoadType is an enumeration of i18n loading types
// +kubebuilder:validation:Enum:=Preload;Lazy

**Comment by jhadvig** [comment]
@yapei issue should be fixed by https://github.com/openshift/console-operator/pull/684

**Comment by yapei** [comment]
above reported issue has been fixed in openshift/console-operator#684, no other issues found
/label qe-approved

**Comment by jhadvig** [comment]
Screenshot 2022-10-20 at 13 37 55

/retest

**Comment by jhadvig** [comment]
Screenshot 2022-10-25 at 22 55 45

**Pull request merged**